src/main/java/de/tum/in/www1/artemis/domain/Exercise.java

```diff
@@ -921,4 +922,63 @@ public String getMappedColumnName() {
         }
     }
 
+    /**
+     * This method is used to validate the dates of an exercise
+     * @throws BadRequestException if the dates are not valid
+     */
+    public void validateDates() throws BadRequestException {
+        // All fields are optional, so there is no error if none of them is set
+        if (getReleaseDate() == null && getDueDate() == null && getAssessmentDueDate() == null) {
+            return;
+        }
+        // atleast one is set, so we have to check the two possible errors
+        boolean validDates = validatedueDateError(getReleaseDate(), getDueDate(), getAssessmentDueDate())
+                && validateAssesmentDueDateError(getReleaseDate(), getDueDate(), getAssessmentDueDate());
+
+        if (!validDates) {
+            throw new BadRequestException();
+        }
+    }
+
+    /**
+     * This method is used to validate the dueDate of an exercise
+     * @return true if the dueDate is valid
+     */
+
+    private boolean validatedueDateError(ZonedDateTime releaseDate, ZonedDateTime dueDate, ZonedDateTime assesmentDueDate) {
+        if (releaseDate == null) {
+            return true;
+        }
+        boolean validDate = true;
+        if (dueDate != null) {
+            validDate = releaseDate.isBefore(dueDate);
+        }
+        if (assesmentDueDate != null && validDate) {
+            validDate = releaseDate.isBefore(dueDate);
+        }
+        return validDate;
```

Comment:
line 957: Did you want to write validDate = dueDate.isBefore(assessmentDueDate) instead ?

It seems like this pattern is repeated throughout this class. What do you think of adding a helper method called `isBeforeDate(Date d1, Date d2)` 
private bool isBeforeDate(Date d1, Date d2) {
    if (d1 == null || d2 == null) return true;
    return d1.isBefore(d2);
}

Then you could write (or something along these lines) and maybe reuse this code in other places in this class:
java
if (releaseDate == null) {
    return true;
}
isBeforeDate(releaseDate, dueDate) && isBeforeDate(dueDate, assessmentDueDate)

Reply:
Thank you! i implemented your advices accordingly